src/hash_map.h

```diff
@@ -50,7 +50,22 @@ unsigned int MurmurHash2(const void* key, size_t len) {
   return h;
 }
 
-#ifdef _MSC_VER
+#if (__cplusplus >= 201103L) || (_MSC_VER >= 1900)
+#include <unordered_map>
+
+namespace std {
+template<>
+struct hash<StringPiece> {
+  typedef StringPiece argument_type;
+  typedef std::size_t result_type;
```

Comment:
Ah, it's in std from cstddef but it's only available globally in stddef.h (like with the other cfoo headers). Ninja uses size_t in an unqualified way everywhere else. Thanks :-)